components/gantry/oneaxis/oneaxis.go

```diff
@@ -358,30 +358,28 @@ func (g *oneAxis) MoveToPosition(ctx context.Context, positions []float64, extra
 			return err
 		}
 
-		// Hits backwards limit switch, goes in forwards direction for two revolutions
-		if hit {
-			if x < g.positionLimits[0] {
-				dir := float64(1)
-				return g.motor.GoFor(ctx, dir*g.rpm, 2, extra)
+		// Hits backwards limit switch, and stops if position x is past the limit switch
+		if (hit == true) || (x <= (g.positionLimits[0] + 0.25)) {
+			g.logger.Debugf("limit: %.2f", g.positionLimits[0]+0.25)
+			g.logger.Debugf("position x: %.2f", x)
+			if x <= (g.positionLimits[0] + 0.25) {
+				g.logger.Errorf("Cannot move past limit switch!")
+				return g.motor.Stop(ctx, extra)
 			}
-			return g.motor.Stop(ctx, extra)
 		}
 
-		// Hits forward limit switch, goes in backwards direction for two revolutions
 		hit, err = g.limitHit(ctx, false)
 		if err != nil {
 			return err
 		}
-		if hit {
-			if x > g.positionLimits[1] {
-				dir := float64(-1)
-				return g.motor.GoFor(ctx, dir*g.rpm, 2, extra)
+		// Hits forwards limit switch, and stops if position x is past the limit switch
+		if (hit == true) || (x >= (g.positionLimits[1] - 0.25)) {
+			g.logger.Debugf("limit: %.2f", g.positionLimits[1]-0.25)
+			g.logger.Debugf("position x: %.2f", x)
+			if x >= (g.positionLimits[1] - 0.25) {
+				g.logger.Errorf("Cannot move past limit switch!")
+				return g.motor.Stop(ctx, extra)
```

Comment:
Same suggestion here. This code and the the other part are nearly identical; I wonder if it would make sense to refactor into a helper function that just gets called twice:
```
func helper(needBetterArgumentNameThanZero, isPastEndPosition bool) err {
    hit, err = g.limitHit(ctx, needBetterArgumentNameThanZero)
    if isPastEndPosition {
        g.logger.Debugf("cannot move past limit switch")
        return g.motor.Stop()
    }
    return nil
}

err = helper(true, x <= g.positionLimits[0] + 0.25)
err = helper(false, x >= g.positionLimits[1] - 0.25)
```
That way, it's very clear which parts are changing and which parts stay the same between the two, and if there are more bugs in the future you only need to fix them once.

Reply:
You can refactor `g.limitHit` to take an offset nd pass that into the helper as well, or refactor limitHit itself to return another bool for thee `x >=positionLimits[idx] +/= 0.25`condition